ui/src/timeMachine/components/builderCard/BuilderCardDropdownHeader.tsx

```diff
@@ -20,17 +21,38 @@ export default class BuilderCardDropdownHeader extends PureComponent<Props> {
   }
 
   public render() {
-    const {children, options, onSelect, selectedOption, testID} = this.props
+    const {
+      children,
+      isInCheckOverlay,
+      options,
+      onSelect,
+      selectedOption,
+      testID,
+    } = this.props
+    const dropdownOptions = isInCheckOverlay ? ['filter'] : options
+    const button = () => (
+      <Dropdown.Button status={ComponentStatus.Disabled} onClick={() => {}}>
+        {selectedOption}
+      </Dropdown.Button>
+    )
 
+    const menu = () => null
     return (
       <div className="builder-card--header" data-testid={testID}>
-        <SelectDropdown
-          options={options}
-          selectedOption={selectedOption}
-          testID="select-option-dropdown"
-          onSelect={onSelect ? onSelect : emptyFunction}
-        />
-
+        {dropdownOptions.length === 1 ? (
+          <Dropdown
+            button={button}
+            menu={menu}
+            testID="disabled--filter-dropdown"
+          />
+        ) : (
+          <SelectDropdown
+            options={dropdownOptions}
+            selectedOption={selectedOption}
+            testID="select-option-dropdown"
+            onSelect={onSelect ? onSelect : emptyFunction}
+          />
+        )}
```

Comment:
```suggestion
        {isInCheckOverlay ? (
          <Dropdown
            button={button}
            menu={menu}
            testID="disabled--filter-dropdown"
          />
        ) : (
          <SelectDropdown
            options={options}
            selectedOption={selectedOption}
            testID="select-option-dropdown"
            onSelect={onSelect ? onSelect : emptyFunction}
          />
        )}
```

following up on the above comment. this is conceptually simpler because it doesn't introduce a new variable. that is the main motivation in asking for this change - i know it's annoying to go back and change a pr like this, but reducing [accidental complexity](https://en.wikipedia.org/wiki/No_Silver_Bullet#Summary) is one of the most valuable things an engineer can spend time on. the complexity of having a new variable (`dropdownOptions`) encode the state of the component, when we already have a variable to do that (`isInCheckOverlay`) is extra complexity that doesn't need to be added.